docs/snippets/angular/my-component-story-with-nonstory.ts.mdx

```diff
@@ -0,0 +1,32 @@
+```ts
+// MyComponent.stories.ts
+
+import { MyComponent } from './MyComponent.component';
+
+import someData from './data.json';
+
+export default {
+  component: MyComponent,
+  includeStories: ['SimpleStory', 'ComplexStory'], // 👈 Storybook loads these stories
+  excludeStories: /.*Data$/, // 👈 Storybook ignores anything that contains Data
+};
+
+export const simpleData = { foo: 1, bar: 'baz' };
+export const complexData = { foo: 1, foobar: { bar: 'baz', baz: someData } };
+
+export const SimpleStory = {
+  render: () => ({
```

Comment:
Same as above and a lot of other places

```js
export const SimpleStory = {
  args: {
    data: simpleData
  }
};
```

Reply:
@shilman I'm aware of the case and let me clarify this a bit, the snippets are using args as much as possible. Such cases still exist due to a couple of reasons:
- Positioning of the snippet in the documentation.
- Maintainability

Let me clarify on the latter. It all boils down to the following question that I have to ask myself when creating examples for :
- React
- Angular
- Vue 2 and 3
- Svelte
- Any other framework

If I can use a unified approach without that can cover the frameworks with a similar pattern that the readers can relate to. I'll use it, with that cutting the number of snippets that are in use and must be maintained and updated while demonstrating the use case. If I cannot then I have to create snippets for each documented framework as I did with the latest commit. I've updated this particular example to use args for the documented frameworks (React, Vue, Angular, Svelte) and split the Vue implementation from one file into two distinct files one for Vue 2 and one for Vue 3. Adding more files/examples to maintain. 

One other question that is constantly "burning in the back of my mind" (pardon the bad pun) is when this approach will become unmaintainable while trying to provide unified versioned documentation. So far we're been able to get a handle on this. But as the documentation grows and grows we have to factor in this.

Reply:
That makes sense. I think once we finish the default `render` functions for all the frameworks it will make the snippets a LOT easier to maintain!